examples/data_selection/influence_function_data_selection.py

```diff
@@ -0,0 +1,151 @@
+"""This example shows how to use the IF to detect low-quality data in the MNIST."""
+
+import argparse
+from functools import partial
+
+import torch
+from torch import nn
+
+from dattri.algorithm.influence_function import (
+    IFAttributorArnoldi,
+    IFAttributorCG,
+    IFAttributorDataInf,
+    IFAttributorExplicit,
+    IFAttributorLiSSA,
+)
+from dattri.benchmark.datasets.mnist import create_mnist_dataset, train_mnist_lr
+from dattri.benchmark.utils import SubsetSampler
+from dattri.metric import mislabel_detection_auc
+from dattri.task import AttributionTask
+
+ATTRIBUTOR_MAP = {
+    "explicit": partial(IFAttributorExplicit, regularization=0.01),
+    "cg": partial(IFAttributorCG, regularization=0.01),
+    "lissa": partial(IFAttributorLiSSA, recursion_depth=100),
+    "datainf": partial(IFAttributorDataInf, regularization=0.01),
+    "arnoldi": partial(IFAttributorArnoldi, regularization=0.01, max_iter=10),
+}
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--method", type=str, default="explicit")
+    parser.add_argument("--device", type=str, default="cuda")
+    parser.add_argument("--seed", type=int, default=0)
+    parser.add_argument("--train_size", type=int, default=10000)
+    parser.add_argument("--val_size", type=int, default=10000)
+    parser.add_argument("--test_size", type=int, default=5000)
+    parser.add_argument("--remove_number", type=int, default=100)
+    args = parser.parse_args()
+
+    print(args)
+
+    # load the training dataset
+    dataset, dataset_test = create_mnist_dataset("./data")
+
+    # for model training, batch size is 64
+    train_loader_full = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=64,
+        sampler=SubsetSampler(range(args.train_size)),
+    )
+
+    # training samples for attribution
+    # batch size is 1000 to speed up the process
+    train_loader = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=1000,
+        sampler=SubsetSampler(range(args.train_size)),
+    )
+    val_loader = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=1000,
+        sampler=SubsetSampler(range(args.train_size, args.train_size + args.val_size)),
+    )
+    test_loader = torch.utils.data.DataLoader(
+        dataset_test,
+        batch_size=1,
```

Comment:
why we suddenly use such a small batch size?

Reply:
well I guess I can make this larger. It does not affect the TDA runtime since only validation dataset are used for TDA.

Reply:
make sense